tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc

```diff
@@ -54,53 +54,191 @@ UnaryOperationMap() {
   return m;
 }
 
-class ConvertUnary : public OpConverterBase<ConvertUnary> {
- public:
-  explicit ConvertUnary(OpConverterParams* params)
-      : OpConverterBase<ConvertUnary>(params) {}
+const unaryOperationMap* UnaryBooleanOperationMap() {
+  static auto* const m = new unaryOperationMap({
+      {"LogicalNot", nvinfer1::UnaryOperation::kNOT},
+  });
+  return m;
+}
 
-  static constexpr std::array<DataType, 3> AllowedDataTypes() {
-    return {DataType::DT_FLOAT, DataType::DT_HALF};
-  }
+const operationMap<nvinfer1::ActivationType>* ActivationTypeMap() {
+  static auto* const m =
+      new std::unordered_map<string, nvinfer1::ActivationType>({
+          {"LeakyRelu", nvinfer1::ActivationType::kLEAKY_RELU},
+          {"Relu", nvinfer1::ActivationType::kRELU},
+          {"Relu6", nvinfer1::ActivationType::kCLIP},
+          {"Sigmoid", nvinfer1::ActivationType::kSIGMOID},
+          {"Tanh", nvinfer1::ActivationType::kTANH},
+          {"Elu", nvinfer1::ActivationType::kELU},
+          {"Selu", nvinfer1::ActivationType::kSELU},
+          {"Softsign", nvinfer1::ActivationType::kSOFTSIGN},
+          {"Softplus", nvinfer1::ActivationType::kSOFTPLUS},
+      });
+  return m;
+}
 
-  static constexpr std::array<InputArgSpec, 1> InputSpec() {
-    return std::array<InputArgSpec, 1>{
-        InputArgSpec::Create("x", TrtInputArg::kTensor)};
-  }
+template <typename T>
+class ConvertUnaryImpl {
+ protected:
+  ConvertUnaryImpl(const operationMap<T>* pOperMap) : pOperMap_(pOperMap) {}
 
-  Status Validate() {
-    const auto& op = params_->node_def.op();
-    if (UnaryOperationMap()->find(op) == UnaryOperationMap()->end()) {
+  Status ImplValidate(const OpConverterParams& params,
+                      const std::vector<string>& not_supported_ops = {}) {
+    const auto& op = params.node_def.op();
+    if (pOperMap_->find(op) == pOperMap_->end()) {
       return errors::Unimplemented("Unary op: ", op, " not supported");
     }
+    DimsAdapter input_dims(params.inputs.at(0).GetTrtDims());
+    if (!input_dims.NumDims()) {
+      return errors::InvalidArgument(
+          "At least 1 dimension is required for UNARY operation '", op, "'");
+    }
 
-    if (params_->use_implicit_batch && (op == "Sign" || op == "Round")) {
-      return errors::Unimplemented("Unary op: '", params_->node_def.op(),
-                                   "' is not supported in implicit batch mode");
+    if (!not_supported_ops.empty() && params.use_implicit_batch) {
+      const auto& end = not_supported_ops.end();
+      if (std::find(not_supported_ops.begin(), end, op) != end) {
+        return errors::Unimplemented(
+            "Unary op: '", op, "' is not supported in implicit batch mode");
+      }
     }
 
     return Status::OK();
   }
 
-  Status Convert() {
-    const auto& params = *this->params_;
+  Status ImplConvert(const OpConverterParams& params) {
```

Comment:
s/ImplConvert/ConvertImpl/ to be consistent with how we use "Impl"?

Reply:
Done